File: NAP II - POO/Java/src/Transportadora.java
import java.util.List;

public class Transportadora {
    private String nome;
    private double precoPorVolume;
    private double precoPorPeso;
    private int prazoEntrega;

    // Construtor padrão
    public Transportadora(String nome, double precoPorVolume, double precoPorPeso, int prazoEntrega) {
        this.nome = nome;
        this.precoPorVolume = precoPorVolume;
        this.precoPorPeso = precoPorPeso;
        this.prazoEntrega = prazoEntrega;
    }

    // Construtor não padrão
    public Transportadora(String nome) {
        this(nome, 0.0, 0.0, 0);
    }

    @Override
    public String toString() {
        return String.format("Transportadora: %s (Preço por Volume: R$ %.2f, Preço por Peso: R$ %.2f, Prazo de Entrega: %d dias)",
                nome, precoPorVolume, precoPorPeso, prazoEntrega);
    }

    // Método para calcular o frete
    public double calcularFrete(List<Item> itens) {
        double freteTotal = 0;
        for (Item item : itens) {
            Produto produto = item.getProduto();
            double precoPorVolume = this.precoPorVolume * produto.getVolume() * item.getQuantidade();
            double precoPorPeso = this.precoPorPeso * produto.getPeso() * item.getQuantidade();
            freteTotal += Math.max(precoPorVolume, precoPorPeso);
        }
        return freteTotal;
    }
}

File: NAP II - POO/Python/UFRAcommerce/transportadora.py
class Transportadora:
    def __init__(self, nome, preco_por_volume, preco_por_peso, prazo_entrega):
        self.nome = nome
        self.preco_por_volume = preco_por_volume
        self.preco_por_peso = preco_por_peso
        self.prazo_entrega = prazo_entrega

    def calcularFrete(self, itens):
        frete_total = 0
        for item in itens:
            custo_por_volume = item.produto.volume * self.preco_por_volume * item.quantidade
            custo_por_peso = item.produto.peso * self.preco_por_peso * item.quantidade
            frete_total += max(custo_por_volume, custo_por_peso)
        return frete_total

    def __str__(self):
        return f"Transportadora(nome={self.nome}, preco_por_volume={self.preco_por_volume}, preco_por_peso={self.preco_por_peso})"
